redbot/cogs/mutes/mutes.py

```diff
@@ -0,0 +1,1219 @@
+        if not can_react:
+            message += " (y/n)"
+        query: discord.Message = await ctx.send(message)
+        if can_react:
+            # noinspection PyAsyncCall
+            start_adding_reactions(query, ReactionPredicate.YES_OR_NO_EMOJIS)
+            pred = ReactionPredicate.yes_or_no(query, ctx.author)
+            event = "reaction_add"
+        else:
+            pred = MessagePredicate.yes_or_no(ctx)
+            event = "message"
+        try:
+            await ctx.bot.wait_for(event, check=pred, timeout=30)
+        except asyncio.TimeoutError:
+            await query.delete()
+            return
+
+        if not pred.result:
+            if can_react:
+                await query.delete()
+            else:
+                await ctx.send(_("OK then."))
+            return
+        else:
+            if can_react:
+                with contextlib.suppress(discord.Forbidden):
+                    await query.clear_reactions()
+            resp = pagify(issue)
+            await ctx.send_interactive(resp)
+
+    @commands.command(
+        name="mutechannel", aliases=["channelmute"], usage="[users...] [time_and_reason]"
+    )
+    @commands.guild_only()
+    @checks.mod_or_permissions(manage_roles=True)
+    async def channel_mute(
+        self,
+        ctx: commands.Context,
+        users: commands.Greedy[discord.Member],
+        *,
+        time_and_reason: MuteTime = {},
+    ):
+        """Mute a user in the current text channel.
+
+        `[users...]` is a space separated list of usernames, ID's, or mentions.
+        `[time_and_reason]` is the time to mute for and reason. Time is
+        any valid time length such as `30 minutes` or `2 days`. If nothing
+        is provided the mute will be indefinite.
+
+        Examples:
+        `[p]mutechannel @member1 @member2 spam 5 hours`
+        `[p]mutechannel @member1 3 days`
+        """
+        if not users:
+            return await ctx.send_help()
+        if ctx.me in users:
+            return await ctx.send(_("You cannot mute me."))
+        if ctx.author in users:
+            return await ctx.send(_("You cannot mute yourself."))
+        if not await self._check_for_mute_role(ctx):
+            return
+        async with ctx.typing():
+            duration = time_and_reason.get("duration", {})
+            reason = time_and_reason.get("reason", None)
+            until = None
+            time = ""
+            if duration:
+                until = datetime.now(timezone.utc) + timedelta(**duration)
+                time = _(" for ") + humanize_timedelta(timedelta=timedelta(**duration))
+            else:
+                default_duration = await self.config.guild(ctx.guild).default_time()
+                if default_duration:
+                    until = datetime.now(timezone.utc) + timedelta(**default_duration)
+                    time = _(" for ") + humanize_timedelta(timedelta=timedelta(**default_duration))
```

Comment:
With this amount of context, it might be hard for translator to translate this properly, maybe something like this would be better? (I'm not sure)
```suggestion
                    duration_str = humanize_timedelta(timedelta=timedelta(**default_duration))
                    time = _(" for {duration}").format(duration=duration_str)
```